core/src/test/scala/org/apache/spark/rpc/RpcEnvSuite.scala

```diff
@@ -624,7 +624,9 @@ abstract class RpcEnvSuite extends SparkFunSuite with BeforeAndAfterAll {
       val e = intercept[SparkException] {
         ThreadUtils.awaitResult(f, 1 seconds)
       }
-      assert(e.getCause.isInstanceOf[NotSerializableException])
+      assert(e.getCause.isInstanceOf[RuntimeException])
```

Comment:
Because we need a `Throwable` meesage called back from server, you can see: [NettyRpcEnv.scala#L205](https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/rpc/netty/NettyRpcEnv.scala#L205)